client/collector/src/main/java/org/hawkular/apm/client/collector/DefaultTraceCollector.java

```diff
@@ -856,19 +856,16 @@ public void setFault(String location, String value, String description) {
     @Override
     public void setProperty(String location, String name, String value) {
         if (log.isLoggable(Level.FINEST)) {
-            log.finest("Set business transaction property: location=" + location +
+            log.finest("Set property: location=" + location +
                     " name=" + name + " value=" + value);
         }
 
         try {
             if (fragmentManager.hasFragmentBuilder()) {
                 FragmentBuilder builder = fragmentManager.getFragmentBuilder();
 
-                if (value == null) {
-                    builder.getTrace().getProperties().remove(name);
-                } else {
-                    builder.getTrace().getProperties().add(new Property(name, value));
-                }
+                builder.getCurrentNode().getProperties().add(new Property(name, value));
```

Comment:
If the `value` is null, the property will be updated to have its value as `null`, while previously the semantics were `remove the entry if the value is null`. Is this what you had in mind on this change?